### PR TITLE
🍖 Hunger : Init

### PR DESCRIPTION
**Pull request by Cobbrou**

### CODE REVIEW

**Review by Guthen — CHANGES_REQUESTED** [review]
See above

**Review by Guthen — APPROVED** [review]

File: gamemodes/virtualroleplay/gamemode/modules/hunger/sv_hunger.lua [hunk 1]
```diff
@@ -0,0 +1,38 @@
+local PLAYER = FindMetaTable( "Player" )
+
+local function HungerThink()
+    if not VRP.HungerEnabled then return end
+    for _, v in ipairs(player.GetAll()) do
+        if not v:Alive() then continue end
+        v:HungerUpdate()
+    end
+end
+timer.Create("HungerThink", VRP.HungerStarvingTime, 0, HungerThink)
```
Comment (Nogitsu) [inline gamemodes/virtualroleplay/gamemode/modules/hunger/sv_hunger.lua:10]: Why are you creating a local function for a one time use ?

File: gamemodes/virtualroleplay/gamemode/modules/hunger/sv_hunger.lua [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+function PLAYER:HungerUpdate()
```
Comment (Nogitsu) [inline gamemodes/virtualroleplay/gamemode/modules/hunger/sv_hunger.lua:12]: Same than before, this function doesn't need to be in the meta, it's a one-time use, only in your "local timer", so you could copy the content direclty in your loop

**Comment by Nogitsu** [comment]
Edited and approved (still need more reviews to confirm my edits)

**Review by Nogitsu — COMMENTED** [review]
Edited and approved (still need more reviews to confirm my edits)

File: gamemodes/virtualroleplay/gamemode/modules/hunger/sv_hunger.lua (hunk 1 above)
Comment (Erlite) [reply gamemodes/virtualroleplay/gamemode/modules/hunger/sv_hunger.lua]: That's fine, I do that too sometimes, cleaner.

File: gamemodes/virtualroleplay/gamemode/modules/hunger/sv_hunger.lua (hunk 2 above)
Comment (Erlite) [reply gamemodes/virtualroleplay/gamemode/modules/hunger/sv_hunger.lua]: That might actually prove useful for modding.

File: gamemodes/virtualroleplay/gamemode/modules/hunger/sv_hunger.lua [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+-- commands
+VRP.AddChatCommand( "sethunger", function( ply, args )
+    if not ply:IsSuperAdmin() then return "You must be a SuperAdmin", 1 end
+
+    local amount = tonumber( args[1] )
+    if not amount or amount < 0 then return "You must specify the amount!", 1 end
+
+    ply:SetHunger( amount )
+    return ( "You set your hunger to %s!" ):format( amount )
```
Comment (Erlite) [inline gamemodes/virtualroleplay/gamemode/modules/hunger/sv_hunger.lua:31]: Might be cleaner to do 
```lua
return string.format( "You set your hunger to %s!", amount )
```

File: gamemodes/virtualroleplay/gamemode/modules/hunger/sv_hunger.lua (hunk 2 above)
Comment (Nogitsu) [reply gamemodes/virtualroleplay/gamemode/modules/hunger/sv_hunger.lua]: I can't see how modding can be usefull in the hunger's case

File: gamemodes/virtualroleplay/gamemode/modules/hunger/sv_hunger.lua [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+end )
+
+
+VRP.AddChatCommand( "resethunger", function( ply, args )
+    if not ply:IsSuperAdmin() then return "You must be a SuperAdmin", 1 end
+
+    local amount = tonumber( args[1] )
+
+    ply:SetHunger( VRP.HungerStartAmount )
+    return ( "You reset your hunger!" )
```
Comment (Erlite) [inline gamemodes/virtualroleplay/gamemode/modules/hunger/sv_hunger.lua:41]: Redundant parentheses
```lua
return "You reset your hunger!"
```

**Review by Erlite — CHANGES_REQUESTED** [review]
Few changes needed

File: gamemodes/virtualroleplay/gamemode/modules/hunger/sv_hunger.lua (hunk 3 above)
Comment (Guthen) [reply gamemodes/virtualroleplay/gamemode/modules/hunger/sv_hunger.lua:31]: I prefer the way it's already done

Comment (Nogitsu) [reply gamemodes/virtualroleplay/gamemode/modules/hunger/sv_hunger.lua:31]: Same

Comment (Erlite) [reply gamemodes/virtualroleplay/gamemode/modules/hunger/sv_hunger.lua:31]: Not as clean or readable, but sure.

**Review by Guthen — APPROVED** [review]
Good, after the merge you should use the new language system for the chat commands. Also, you should remove the parentheses as said Erlite (but actually it should be gone after the language system implementation).

**Review by Nogitsu — APPROVED** [review]

**Review by Erlite — APPROVED** [review]

**Pull request merged**